lib/pure/os.nim

```diff
@@ -3187,38 +3187,54 @@ proc getFileInfo*(path: string, followSymlink = true): FileInfo {.noWeirdTarget.
         raiseOSError(osLastError(), path)
     rawToFormalFileInfo(rawInfo, path, result)
 
-proc sameFileContent*(path1, path2: string): bool {.rtl, extern: "nos$1",
-  tags: [ReadIOEffect], noWeirdTarget.} =
-  ## Returns true if both pathname arguments refer to files with identical
-  ## binary content.
+proc sameFileContent*(path1, path2: string; checkSize = false; checkFiles = false; bufferSize = 0.Natural): bool {.
+    rtl, extern: "nos$1", tags: [ReadIOEffect], noWeirdTarget.} =
+  ## Returns `true` if both pathname arguments refer to files with identical binary content.
+  ##
+  ## If `checkSize` is `true` then checks the file sizes *before reading the files*,
+  ## if the files have different file sizes they can not have the same contents,
+  ## `checkSize = true` may be faster, specially for very big files.
+  ## This does not fail if the file never existed in the first place, unless `checkFiles = true`.
   ##
   ## See also:
   ## * `sameFile proc <#sameFile,string,string>`_
-  var
-    a, b: File
-  if not open(a, path1): return false
-  if not open(b, path2):
+  runnableExamples:
+    doAssert sameFileContent(currentSourcePath, currentSourcePath, checkSize = true, bufferSize = 4096)
+  var a, b: File
+  var mustRead = true
+  var bufA = alloc(bufferSize)
```

Comment:
how can that work? that's a bug.
the way i suggested in https://github.com/nim-lang/Nim/pull/16008#discussion_r528063361 works